src/EventListener/NotificationListener.php

```diff
@@ -0,0 +1,292 @@
+     */
+    private $entityManager;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * NotificationListener constructor.
+     * @param EntityManagerInterface $entityManager
+     * @param RequestStack $requestStack
+     */
+    public function __construct(EntityManagerInterface $entityManager, RequestStack $requestStack)
+    {
+        $this->entityManager = $entityManager;
+        $this->requestStack = $requestStack;
+    }
+
+
+    public function onLectureCreate(LectureEvent $event){
+
+        $lecture = $event->getLecture();
+        $course = $lecture->getCourse();
+        $teacher = $lecture->getTeacher();
+
+        foreach($course->getCourseUsers() as $courseUser){
+            if($courseUser->getRole() == RoleInterface::STUDENT &&
+                $courseUser->getStatus() == StatusInterface::ACTIVE){
+                $notification = new Notification();
+
+                $notification->setUser($courseUser->getUser());
+                $notification->setCourse($course);
+
+                $message = "A new lecture \"".$lecture->getTitle()."\" has been posted
+                in course ".$course->getTitle()." by ".$teacher->getName()." "
+                    .$teacher->getSurname();
+
+                $notification->setMessage($message);
+                $notification->setDate(new \DateTime());
+                $notification->setIsAcceptable(false);
+
+                $this->entityManager->persist($notification);
+            }
+        }
+        $this->entityManager->flush();
+    }
+
+    public function onLectureEdit(LectureEvent $event){
+
+        $lecture = $event->getLecture();
+        $course = $lecture->getCourse();
+        $teacher = $lecture->getTeacher();
+
+        foreach($course->getCourseUsers() as $courseUser){
+            if($courseUser->getRole() == RoleInterface::STUDENT &&
+                $courseUser->getStatus() == StatusInterface::ACTIVE){
+                $notification = new Notification();
+
+                $notification->setUser($courseUser->getUser());
+                $notification->setCourse($course);
+
+                $message = "A lecture \"".$lecture->getTitle()."\" has been edited
+                in course ".$course->getTitle()." by ".$teacher->getName()." "
+                    .$teacher->getSurname();
+
+                $notification->setMessage($message);
+                $notification->setDate(new \DateTime());
+                $notification->setIsAcceptable(false);
+
+                $this->entityManager->persist($notification);
+            }
+        }
+        $this->entityManager->flush();
+    }
+
+    public function onAssignmentCreate(AssignmentEvent $event){
+
+        $assignment = $event->getAssignment();
+        $course = $assignment->getCourse();
+        $teacher = $assignment->getTeacher();
+
+        foreach($course->getCourseUsers() as $courseUser){
+            if($courseUser->getRole() == RoleInterface::STUDENT &&
```

Comment:
Kas be ko, tokias salygas, kurios eina per keles eilutes mes irgi iskeliam i atskira metoda, nes galas supras ka tu cia nori pasakyt kol gilintis nepradedi. Cia turetu buti iskelta i kazka panasaus kaip `isActiveStudent` ar kazkas tokio